content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,724 @@
+```
+
+Check the Deployment:
+```shell
+kubectl get deployment configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-env-var   3/3     3            3           7s
+```
+
+Check the pods for this Deployment (matching by
+{{< glossary_tooltip text="selector" term_id="selector" >}}):
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                                 READY   STATUS    RESTARTS   AGE
+configmap-env-var-59cfc64f7d-74d7z   1/1     Running   0          46s
+configmap-env-var-59cfc64f7d-c4wmj   1/1     Running   0          46s
+configmap-env-var-59cfc64f7d-dpr98   1/1     Running   0          46s
+```
+
+The key-value pair in the ConfigMap is configured as an environment variable in the container of the Pod.
+Check this by viewing the logs of one Pod that belongs to the Deployment.
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+You should see an output similar to:
+```
+Found 3 pods, using pod/configmap-env-var-7c994f7769-l74nq
+Thu Jan  4 16:07:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:26 UTC 2024 The basket is full of apples
+```
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap fruits
+```
+
+In the editor that appears, change the value of key `fruits` from `apples` to `mangoes`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+```yaml
+apiVersion: v1
+data:
+  fruits: mangoes
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-04T16:04:19Z"
+  name: fruits
+  namespace: default
+  resourceVersion: "1749472"
+
+```
+
+You should see the following output:
+```
+configmap/fruits edited
+```
+
+Tail the logs of the Deployment and observe the output for few seconds:
+```shell
+# As the text explains, the output does NOT change
+kubectl logs deployments/configmap-env-var --follow
+```
+
+Notice that the output remains **unchanged**, even though you edited the ConfigMap:
+```
+Thu Jan  4 16:12:56 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:26 UTC 2024 The basket is full of apples
+```
+
+{{< note >}}
+Although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still shows the earlier value. This is because environment variables for a process running inside a Pod are **not** updated when the source data changes; if you wanted to force an update, you would need to have Kubernetes replace your existing Pods. The new Pods would then run with the updated information.
+{{< /note >}}
```

Comment:
Since this subsection only differs from the previous one on how the changes are effected, I'd suggest we focus on the differences. Even for a tutorial, we don't need to enumerate each and every steps.

Reply:
I believe it will help the reader to copy/paste the commands for ease of learning. So I propose to keep the steps..

Reply:
Thanks, I've made the necessary changes.

Reply:
If you want to keep this subsection rather than drop it completely, you may want to focus on the "meaningful" steps, which is that users have to recreate their pods. There are many ways to recreate them. The simplest one is to delete and recreate.
I mean we don't need the lines on creating and verifying the ConfigMap or the Deployment.

Reply:
The deployment manifest in the section for ConfigMap as environment variable is different to the previous section where the ConfigMap is mounted as a Volume. So I think it is necessary to provide the exact manifest on how to create the deployment, so that the reader does not make a mistake here.